### PR TITLE
Add a comments section to the page.

### PR DESCRIPTION
**Pull request by officialcwang**
User can input comments, which will then display beneath the text box. 
Looks like: https://i.imgur.com/Ca3Uzpd.jpg

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -26,21 +26,40 @@
...
+    String json = gson.toJson(comments);
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(greetings);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (AnudeepYakkala) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: Add a comment describing what the function does.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,21 +26,40 @@
...
     response.setContentType("text/html;");
     response.getWriter().println(json);
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (AnudeepYakkala) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: Dawn told me to use an optional for this function. 

Optional<String> getParameter(HttpServletRequest request, String param)
Effective Java Item 55: Return optionals judiciously

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -5,15 +5,23 @@
...
       <p class="description">This is Carolyn Wang.</p>
       <img src="/images/3d.jpeg" alt="3d-glasses" class="photo" />
-      <div class="button">
-        <button type="button" onclick="addHello()">Say hello</button>
+      <form action="/data" method="post">
+        <p>Enter a comment:</p>
+        <textarea name="text-input"></textarea>
+        <br />
+        <div class="button">
+          <button type="submit">Say hello</button>
```
Comment (AnudeepYakkala) [inline portfolio/src/main/webapp/index.html:18]: Change the button to say "Comment" or something similar.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -15,17 +15,26 @@
 /* exported addRandomQuote */
 /* exported randomizeImage */
 /* exported createNavigation */
-/* exported addHello */
+/* exported addComment */
 
 /**
  * Fetches hello from the server and adds it to the page.
```
Comment (AnudeepYakkala) [inline portfolio/src/main/webapp/script.js:21]: Change comment.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (officialcwang) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: Thanks for letting me know!

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (officialcwang) [reply portfolio/src/main/webapp/index.html:18]: Eh, I think I like it being something different and less impersonal.

Comment (AnudeepYakkala) [reply portfolio/src/main/webapp/index.html:18]: Oh, I thought you forgot to change it from a previous version. My bad.

**Pull request merged**